File: library/src/test/java/me/proxer/library/api/info/ModifyUserInfoEndpointTest.java
package me.proxer.library.api.info;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import okhttp3.mockwebserver.MockResponse;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Ruben Gees
 */
class ModifyUserInfoEndpointTest extends ProxerTest {

    @Test
    void testDefault() throws ProxerException, IOException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        final Void result = api.info()
                .note("123")
                .build()
                .execute();

        assertThat(result).isNull();
    }

    @Test
    void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.info().note("321")
                .build()
                .execute();

        assertThat(server.takeRequest().getPath()).isEqualTo("/api/v1/info/setuserinfo");
    }

    @Test
    void testNoteParameter() throws IOException, ProxerException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.info().note("321")
                .build()
                .execute();

        assertThat(server.takeRequest().getBody().readUtf8()).isEqualTo("id=321&type=note");
    }

    @Test
    void testFavoriteParameter() throws IOException, ProxerException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.info().markAsFavorite("321")
                .build()
                .execute();

        assertThat(server.takeRequest().getBody().readUtf8()).isEqualTo("id=321&type=favor");
    }

    @Test
    void testFinishedParameter() throws IOException, ProxerException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.info().markAsFinished("321")
                .build()
                .execute();

        assertThat(server.takeRequest().getBody().readUtf8()).isEqualTo("id=321&type=finish");
    }
}

File: library/src/test/kotlin/me/proxer/library/api/info/ModifyUserInfoEndpointTest.kt
package me.proxer.library.api.info

import me.proxer.library.ProxerTest
import me.proxer.library.runRequest
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldBeNull
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class ModifyUserInfoEndpointTest : ProxerTest() {

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("empty.json") {
            api.info
                .note("123")
                .build()
                .execute()
        }

        result.shouldBeNull()
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("empty.json") {
            api.info.note("321")
                .build()
                .execute()
        }

        request.path shouldBeEqualTo "/api/v1/info/setuserinfo"
    }

    @Test
    fun testNoteParameter() {
        val (_, request) = server.runRequest("empty.json") {
            api.info.note("321")
                .build()
                .execute()
        }

        request.body.readUtf8() shouldBeEqualTo "id=321&type=note"
    }

    @Test
    fun testFavoriteParameter() {
        val (_, request) = server.runRequest("empty.json") {
            api.info.markAsFavorite("321")
                .build()
                .execute()
        }

        request.body.readUtf8() shouldBeEqualTo "id=321&type=favor"
    }

    @Test
    fun testFinishedParameter() {
        val (_, request) = server.runRequest("empty.json") {
            api.info.markAsFinished("321")
                .build()
                .execute()
        }

        request.body.readUtf8() shouldBeEqualTo "id=321&type=finish"
    }
}
